### PR TITLE
Command line interface and a bunch of insignificant changes/improvements.

### PR DESCRIPTION
**Pull request by moorchegue**
Diff seems pretty big, but it's just some castling around, the only relatively big thing is CLI.

### CODE REVIEW

**Comment by Jiehong** [comment]
Hello!

Thank you very much for this. I will have a thorough look at it very soon!

File: PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Maintainer: murchik <mixturchik@gmail.com>
+pkgname=zhudi-cli
```
Comment (Jiehong) [inline PKGBUILD:2]: Not sure this PKGBUILD is needed for zhudi, since it provides more than that.

**Comment by Jiehong** [comment]
Other than the PKGBUILD, It's good to me. I can include your PR as-is, and re-organise a little bit the PKGBUILD, if that's ok to you.

**Comment by moorchegue** [comment]
Sure. That PKGBUILD I added later after creating a pull request to put this to AUR as well. You can totally wipe it out.

P.S. After using it in CLI mode for a while I ran into some more bugs, but ain't had time to fix it yet. Hopefully there'll be more pull requests.

**Pull request merged**